### PR TITLE
Active section for navigation bar

### PR DESCRIPTION
**Pull request by avinhurry**
### Context

- Logic for the active section on the Navigation bar which changes depending on which page you are on (Home or Trainee records)

### Changes proposed in this pull request

- Add logic to navigation bar component to render the blue bar on the active tab

### Guidance to review

- Pull down the code
- Sign in 
- Check that the blue bar is on the 'Home' tab within the navigation bar
- Click on the 'Trainee records' tab and check the blue bar changes dynamically

### Screenshots

image

image

### CODE REVIEW

File: app/components/navigation_bar/view.html.erb [hunk 1]
```diff
@@ -4,12 +4,15 @@
       <div class="moj-primary-navigation__nav">
         <nav class="moj-primary-navigation" aria-label="Primary navigation">
           <ul class="moj-primary-navigation__list">
+          <% items.each do |item| %>
             <li class="moj-primary-navigation__item">
-              <%= govuk_link_to "Home", home_path, { class: "moj-primary-navigation__link" } %>
-            </li>
-            <li class ="moj-primary-navigation__item">
-              <%= govuk_link_to "Trainee records", trainees_path, { class: "moj-primary-navigation__link" } %>
+              <% if item.fetch(:current, false) || current_page?(item.fetch(:url)) %>
```
Comment (AbigailMcP) [inline app/components/navigation_bar/view.html.erb:9]: So this gives us the ability to pass in an item like `{ name: "Tab name", url: tab_path, current: true }` (with current = true),w and override it being rendered as active even though its url doesn't match the `current_url`.

You're not actually using this anywhere in the codebase though - the items are never supplied with `current: true`.

**Review by AbigailMcP — COMMENTED** [review]
Looks good so far!

This works for when you're on the homepage and the trainee records page, since the urls match the urls of the navigation links.

However, when you move to other pages like the [individual trainee pages](https://register-prototype.herokuapp.com/record/7603ff1e-8bdb-46cb-9621-6ed3f1867cd1), the active styling disappears. The blue line should be under 'Trainee records'.

Maybe you could use your `current: true` functionality?

**Comment by avinhurry** [comment]
[Active section for navigation bar](https://trello.com/c/BTdxPnJo/888-active-section-for-navigation-bar)

**Comment by avinhurry** [comment]
This PR was closed and replaced by: https://github.com/DFE-Digital/register-trainee-teachers/pull/408

**Pull request closed without merging**